proto.Merge broken on intstr in Kubernetes 1.16 apimachinary

**What happened**:
I have a test:
```go
func TestIntstr(t *testing.T) {
	x := intstr.Parse("1")
	proto.Merge(&x, &x)
}
```

That passes with `k8s.io/apimachinery v0.0.0-20190612205821-1799e75a0719` (Kubernetes 1.15) but fails with `k8s.io/apimachinery v0.0.0-20190913080033-27d36303b655` (Kubenernetes 1.16)

Logs from pass:
```
=== RUN   TestIntstr
2019-10-14T18:11:33.446269Z	info	proto: don't know how to copy 0
--- PASS: TestIntstr (0.00s)
PASS
```

Logs from fail:
```
=== RUN   TestIntstr
--- FAIL: TestIntstr (0.00s)
panic: merger not found for type:intstr.Type [recovered]
	panic: merger not found for type:intstr.Type

goroutine 18 [running]:
testing.tRunner.func1(0xc00038a100)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x10b4120, 0xc0002d81c0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo(0xc0002de680)
	/usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:662 +0xfb3
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc0002de680, 0xc0002d23c0, 0xc0002d23c0)
	/usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:113 +0x4a2
github.com/gogo/protobuf/proto.(*InternalMessageInfo).Merge(0x1e37f00, 0x1467fe0, 0xc0002d23c0, 0x1467fe0, 0xc0002d23c0)
	/usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/table_merge.go:50 +0x4c
k8s.io/apimachinery/pkg/util/intstr.(*IntOrString).XXX_Merge(0xc0002d23c0, 0x1467fe0, 0xc0002d23c0)
	/usr/local/google/home/howardjohn/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/intstr/generated.pb.go:60 +0x57
github.com/gogo/protobuf/proto.Merge(0x1467fe0, 0xc0002d23c0, 0x1467fe0, 0xc0002d23c0)
	/usr/local/google/home/howardjohn/go/pkg/mod/github.com/gogo/protobuf@v1.3.0/proto/clone.go:95 +0x327
```

`proto` here is `"github.com/gogo/protobuf/proto"`. If I switch to `"github.com/golang/protobuf/proto"` this passes on both version.

**What you expected to happen**:
`proto.Merge` does not panic.

**How to reproduce it (as minimally and precisely as possible)**:
Test and apimachinary version above

Reproducer at https://github.com/liggitt/protobuf/commits/unsized-merge

```
$ go test ./test/merge/
--- FAIL: TestMerge6 (0.00s)
panic: merger not found for type:merge.CustomUnsizedType [recovered]
	panic: merger not found for type:merge.CustomUnsizedType

goroutine 24 [running]:
testing.tRunner.func1(0xc0000e5500)
	/Users/liggitt/.gvm/gos/go1.12.10/src/testing/testing.go:830 +0x392
panic(0x11ea940, 0xc0000dc0d0)
	/Users/liggitt/.gvm/gos/go1.12.10/src/runtime/panic.go:522 +0x1b5
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo(0xc0000da300)
	/Users/liggitt/go/src/github.com/gogo/protobuf/proto/table_merge.go:662 +0xf82
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc0000da300, 0xc0000e0270, 0xc0000e02a0)
	/Users/liggitt/go/src/github.com/gogo/protobuf/proto/table_merge.go:113 +0x4b0
github.com/gogo/protobuf/proto.(*InternalMessageInfo).Merge(0x13fe1e0, 0x127a940, 0xc0000e0270, 0x127a940, 0xc0000e02a0)
	/Users/liggitt/go/src/github.com/gogo/protobuf/proto/table_merge.go:50 +0x4f
github.com/gogo/protobuf/test/merge.(*D).XXX_Merge(0xc0000e0270, 0x127a940, 0xc0000e02a0)
	/Users/liggitt/go/src/github.com/gogo/protobuf/test/merge/merge.pb.go:167 +0x57
github.com/gogo/protobuf/proto.Merge(0x127a940, 0xc0000e0270, 0x127a940, 0xc0000e02a0)
	/Users/liggitt/go/src/github.com/gogo/protobuf/proto/clone.go:95 +0x32d
github.com/gogo/protobuf/test/merge.TestMerge6(0xc0000e5500)
	/Users/liggitt/go/src/github.com/gogo/protobuf/test/merge/merge_test.go:74 +0xc5
testing.tRunner(0xc0000e5500, 0x1243918)
	/Users/liggitt/.gvm/gos/go1.12.10/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/Users/liggitt/.gvm/gos/go1.12.10/src/testing/testing.go:916 +0x35a
FAIL	github.com/gogo/protobuf/test/merge	0.070s
```

--- a/staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go
+++ b/staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go
@@ -45,7 +45,7 @@ type IntOrString struct {
 }
 
 // Type represents the stored type of IntOrString.
-type Type int
+type Type int64
 
 const (
 	Int    Type = iota // The IntOrString holds an int.

